src/proto/faabric.proto

```diff
@@ -84,6 +84,10 @@ message MPIMessage {
     bytes buffer = 8;
 }
 
+message MpiHostRankMsg {
```

Comment:
Note that `protobuf` [guarantees in-order delivery for `repeated` fields](https://developers.google.com/protocol-buffers/docs/proto3#specifying_field_rules).